book/src/languages.md

```diff
@@ -73,6 +73,7 @@ The `language-server` field takes the following keys:
 | `args`        | A list of arguments to pass to the language server binary             |
 | `timeout`     | The maximum time a request to the language server may take, in seconds. Defaults to `20` |
 | `language-id` | The language name to pass to the language server. Some language servers support multiple languages and use this field to determine which one is being served in a buffer |
+| `envs`        | Any environment variables that will be used when starting the language server as an array of key/value pairs [["Key", "Value"], ["Key", "Value"]] |
```

Comment:
Why not a `HashMap<String, String>` which would be a toml table?

```toml
language-server = { command = "mylang-lsp", args = [], env = { ENV1 = "value1", ENV2 = "value2" } }
```

This would also make it possible to write the env more clearly when there are many:

```toml
[[language]]
name = "mylang"

[language.language-server] 
command = "mylang-lsp"
args = []

[language.language-server.env]
ENV1 = "value1"
ENV2 = "value2"
# ...
```

Reply:
Also, this is a bit of a nitpick but I'd prefer `environment` which is mode readable

Reply:
That's a good point, and yes I think it would be better!